### PR TITLE
6 bytes protocol

### PR DESCRIPTION
**Pull request by Meli-0xFF**
Done

### CODE REVIEW

File: ball_detect/main.c [hunk 1]
```diff
@@ -76,25 +81,29 @@ void setup(void) {
...
-			modulo = vision_result[j];
-			position = 1000000;
-			for(i = 0; i < 7; i++) {
-				sendc((modulo / position) % 10);
-				modulo %= position;
-				position /= 10;
-			}
+		if((address_of_message % 2) == 0) {
+			now = vision_result[address_of_message / 2];
+			message = now & 0b11111111;
```
Comment (pepol) [inline ball_detect/main.c]: I'd prefer to use hexadecimal notation here.

**Comment by pepol** [comment]
@Meli-0xFF apart from that one comment, I'd prefer it if you put the numbers used to constants, to give them more meaning.

File: ball_detect/main.c [hunk 2]
```diff
@@ -17,14 +17,22 @@
...
-volatile uint8_t pinstate, ct, changed_bits, portb_history = 0xFF, my_address;
+volatile int32_t not_seeing[QUANTITY_OF_SENSORS];
+volatile int32_t ends_of_pulses[QUANTITY_OF_SENSORS];
+volatile int32_t lenghts_of_pulses[QUANTITY_OF_SENSORS];
+volatile int32_t starts_of_pulses[QUANTITY_OF_SENSORS]; 
+volatile int32_t vision_result[QUANTITY_OF_SENSORS];
+volatile int32_t sensor_values[QUANTITY_OF_SENSORS];
+volatile int32_t counter[QUANTITY_OF_SENSORS];
+volatile uint32_t now;
+volatile uint8_t pinstate, ct, changed_bits, portb_history = 0xFF, my_address, address_of_message = 0;
```
Comment (pepol) [inline ball_detect/main.c]: I'm not sure but I think this message might be more than 80 characters long. Could you please check?

File: ball_detect/main.c [hunk 3]
```diff
@@ -76,25 +84,29 @@ void setup(void) {
...
+			message = now & 0xFF;
+			now >>= 8;
+			now &= 0xFF;
+		} else {
+			message = now;
+		}
+		if(address_of_message == 5) {
+			address_of_message = 0;
+		} else {
+			address_of_message++;
```
Comment (pepol) [inline ball_detect/main.c]: How about replacing this whole `if` with `address_of_message = (address_of_message + 1) % 6`? Did I get it right?

File: ball_detect/main.c [hunk 4]
```diff
@@ -103,24 +115,34 @@ ISR(PCINT0_vect) {
...
+			if (pinstate & (1 << ct)) { 
 				starts_of_pulses[ct] = TCNT1; 
 			} else {
 				ends_of_pulses[ct] = TCNT1;
-				lenghts_of_pulses[ct] = abs(ends_of_pulses[ct] - starts_of_pulses[ct]);
+				if(ends_of_pulses[ct] < starts_of_pulses[ct]) {
+					ends_of_pulses[ct] += TIMER_OVERFLOW;
+				}
+				lenghts_of_pulses[ct] = ends_of_pulses[ct] -
+							starts_of_pulses[ct];
```
Comment (pepol) [inline ball_detect/main.c]: Please put the minus (`-`) sign on the line 126 and align it with `=`:

``` cpp
lenghts = ends
        - starts;
```

**Comment by pepol** [comment]
@Meli-0xFF generally looks good, please address those comments above though. Also, is this code tested?

**Comment by pepol** [comment]
@Meli-0xFF generally looks good, please address those comments above though. Also, is this code tested?

**Comment by pepol** [comment]
@Meli-0xFF generally looks good, please address those comments above though. Also, is this code tested?

**Comment by Meli-0xFF** [comment]
Yup, code is tested :)

File: ball_detect/main.c [hunk 5]
```diff
@@ -12,115 +12,134 @@
...
+    //LED REGISTER SETUP
+    DDRD |= 1 << 3;
 
-	//TIMER SETUP
-	TCCR1A = 0;
-	TCCR1B = 2;
-	TCNT1 = 0;
+    //DIP-SWITCH INITIALIZATION
+    DDRC = 0;
+    PORTC |= ((1 << DIP_SWITCH_1)
```
Comment (pepol) [inline ball_detect/main.c]: Please fix alignment as discussed offline.

File: ball_detect/main.c [hunk 6]
```diff
@@ -12,115 +12,134 @@
...
-		my_address = 3;
-	} else if((PINC & (1 << DIP_SWITCH_4)) == 0) {
-		my_address = 4;
-	}
-	
-	//TURN ON INTERRUPTS
-	sei();
+    //INTERRUPTS INIT
+    DDRB = 0;
+    PORTB |= ((1 << BALL_SENSOR_1)
```
Comment (pepol) [inline ball_detect/main.c]: Please fix alignment as discussed offline.

File: ball_detect/main.c [hunk 7]
```diff
@@ -12,115 +12,134 @@
...
+        if (changed_bits & (1 << ct)) {
+            if (pinstate & (1 << ct)) {
+                starts_of_pulses[ct] = TCNT1;
+            } else {
+                ends_of_pulses[ct] = TCNT1;
+                if(ends_of_pulses[ct] < starts_of_pulses[ct]) {
+                    ends_of_pulses[ct] += TIMER_OVERFLOW;
+                }
+                lenghts_of_pulses[ct] = ends_of_pulses[ct]
+                                      -    starts_of_pulses[ct];
```
Comment (pepol) [inline ball_detect/main.c]: Please remove the spaces between the minus sign (`-`) and `starts_of_pulses[ct];`.

File: ball_detect/main.c [hunk 8]
```diff
@@ -12,115 +12,134 @@
...
-	while (1) {
-		vision_result[0] = lenghts_of_pulses[0];
-		vision_result[1] = lenghts_of_pulses[1];
-		vision_result[2] = lenghts_of_pulses[2];
-	}
-	return 0;
+    int i;
+    setup();
+    while (1) {
+        for(i = 0; i < 3; i++) {
```
Comment (pepol) [inline ball_detect/main.c]: Why is there `3` here?

**Comment by pepol** [comment]
LGTM, please squash.

**Comment by pepol** [comment]
Obsoleted-by: #8

**Pull request closed without merging**